testing/f_test.go

```diff
@@ -0,0 +1,91 @@
+package testing
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"testing"
+)
+
+// addString adds a string to the input vector corresponding to consumer.go @ go-fuzz-headers
+func addString(input []byte, s string) []byte {
+	input = binary.BigEndian.AppendUint32(input, uint32(len(s))) // Add a uint32 length
+	input = append(input, []byte(s)...)                          // Add string
+	return input
+}
+
+// addBytes adds a []byte to the input vector corresponding to consumer.go @ go-fuzz-headers
+func addBytes(input []byte, data []byte) []byte {
+	input = binary.BigEndian.AppendUint32(input, uint32(len(data))) // Add a uint32 length
+	input = append(input, data...)                                  // Add string
+	return input
+}
+
+// addU64 adds a uint64 to the input vector corresponding to consumer.go @ go-fuzz-headers
+func addU64(input []byte, i uint64) []byte {
+	input = binary.BigEndian.AppendUint64(input, i)
+	input = append(input, 1) // endianness boolean
+	return input
+}
+
+func addU16(input []byte, i uint64) []byte {
+	input = binary.BigEndian.AppendUint16(input, uint16(i))
+	input = append(input, 1) // endianness boolean
+	return input
```

Comment:
```suggestion
	return append(input, 1) // endianness boolean
```
goes for all these realy